core/Pos/Aeson/Core.hs

```diff
@@ -39,14 +49,56 @@ deriving instance ToJSON SlotCount
 instance FromJSON CoinPortion where
     parseJSON v = unsafeCoinPortionFromDouble <$> parseJSON v
 
+instance ToJSON CoinPortion where
+    toJSON = toJSON . coinPortionToDouble
+
 deriveFromJSON S.defaultOptions ''VssCertificate
 deriveFromJSON S.defaultOptions ''Millisecond
 deriveFromJSON S.defaultOptions ''Microsecond
 deriveFromJSON S.defaultOptions ''Second
 deriveFromJSON S.defaultOptions ''SoftforkRule
 deriveFromJSON S.defaultOptions ''BlockVersionData
+deriveToJSON   S.defaultOptions ''Microsecond
 
 deriving instance FromJSON Timestamp
+deriving instance ToJSON Timestamp
+
+-- Instances needed for wallet
+deriveFromJSON defaultOptions ''SlotId
+deriveFromJSON defaultOptions ''BlockCount
+deriveFromJSON defaultOptions ''LocalSlotIndex
+deriveFromJSON defaultOptions ''ChainDifficulty
+
+instance ToJSON Script where
+    toJSON Script{..} = object [
+        "scrVersion"    .= scrVersion,
+        "scrScript" .= JsonByteString scrScript ]
+
+instance FromJSON Script where
+    parseJSON = withObject "Script" $ \o -> do
```

Comment:
I don't think it's a good idea to call a variable `o`, because in some editors/viewers it can look similar to `0`. Not important, just saying.